dso-l2/src/main/java/com/tc/objectserver/entity/EntityManagerImpl.java

```diff
@@ -103,7 +106,20 @@ public void enterActiveState() {
     // We can promote directly because this method is only called from PTH initialize 
     //  thus, this only happens once RTH is spun down and PTH is beginning to spin up.  We know the request queues are clear
     try {
-      for (ManagedEntity entity : this.entities.values()) {
+      // issue-439: We need to sort these entities, ascending by consumerID.
+      List<ManagedEntity> sortingList = new ArrayList<ManagedEntity>(this.entities.values());
+      Collections.sort(sortingList, new Comparator<ManagedEntity>() {
+        @Override
+        public int compare(ManagedEntity o1, ManagedEntity o2) {
+          long firstID = o1.getConsumerID();
+          long secondID = o2.getConsumerID();
+          return (firstID > secondID)
+              ? 1
+              : (firstID < secondID)
+                ? -1
+                : 0;
+        }});
```

Comment:
As in, assert that the values are NOT equal?  I suppose that would make sense since each of these should be unique.